addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3931,6 +3931,94 @@ options.registry.GalleryElement = options.Class.extend({
     },
 });
 
+options.registry.Button = options.Class.extend({
+    forceDuplicateButton: true,
+
+    /**
+     * @override
+     */
+    onBuilt(options) {
+        // Only if the button is built, not if a snippet containing that button
+        // is built (e.g. true if dropping a button from the snippet menu onto
+        // the page, false if dropping an "image-text" snippet).
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
+    /**
+     * @override
+     */
+    onClone(options) {
+        // Only if the button is cloned, not if a snippet containing that button
+        // is cloned.
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Checks if there are buttons before or after the target element and
+     * applies appropriate styling.
+     *
+     * @private
+     */
+    _adaptButtons() {
+        const previousSiblingEl = this.$target[0].previousElementSibling;
+        const nextSiblingEl = this.$target[0].nextElementSibling;
+        let buttonNeighbor = false;
+        // When multiple buttons follow each other, they may break on 2 lines or
+        // more on mobile, so they need a margin-bottom. Also, if the button is
+        // dropped next to another button add a space between them.
+        if (previousSiblingEl?.matches(".btn")) {
+            previousSiblingEl.classList.add("mb-2");
+            buttonNeighbor = true;
+            this.$target[0].insertAdjacentHTML('beforebegin', '&nbsp;&nbsp;');
```

Comment:
In fact this will prevent me to merge because ci/security. I'll change to using before and after function and using one space instead of 2 non breaking ones. I don't understand why we needed two non breaking ones actually?